airflow/models/taskinstance.py

```diff
@@ -3398,10 +3405,12 @@ def clear_db_references(self, session):
         for table in tables:
             session.execute(
                 delete(table).where(
-                    table.dag_id == self.dag_id,
-                    table.task_id == self.task_id,
-                    table.run_id == self.run_id,
-                    table.map_index == self.map_index,
+                    # Note: Need to add some type ignores, mypy does not understand that all tables
```

Comment:
Hmm. I think that if all-else-fails we can add .pyi files for those ?

Reply:
Okay, added a second commit which adds a bit of re-factoring by introduction of a base class... don't know if this is better or making it worse...